Migrate superflare and @superflare/remix from remix v2 → react-router v7

acusti opened the pull request:
this also meant migrating `examples/remix-cms` from the v1 file routing style to the v2 flat routing style (1296744) and removing the deprecated `@remix-run/eslint-config` package from `apps/site` and `examples/remix-cms`.

the commit that migrates superflare and @superflare/remix is baca18b. doing so required duplicating @react-router/dev’s [`node-adapter.js`](https://github.com/remix-run/react-router/blob/main/packages/react-router-dev/vite/node-adapter.ts) and maintaining it within superflare-remix, because it’s no longer included in the published package’s `dist/` directory. i also made a [PR in react-router](https://github.com/remix-run/react-router/pull/12774) that exposes the `(from|to)NodeRequest` utilities that we depend on in the `superflareDevProxyVitePlugin` and would remove the need for duplicating that file in superflare. if it gets merged, we can remove `superflare-remix/node.adapter.ts` and import those utils alongside `cloudflareDevProxy`. note that i also needed to update superflare-remix’s `tsconfig.json` `module` and `moduleResolution` settings to get it to build without error (presumably due to changes between the published remix vs react-router packages).

acusti committed:
Cleanup additional defunct JSON config flags

follow-up to 905ba4c6cf43de5be83eca5960ff4e182a486152

acusti committed:
Migrate to single fetch (prep for rr v7)

https://remix.run/docs/en/main/guides/single-fetch

acusti committed:
Migrate examples/remix-cms to RR v7

acusti committed:
Migrate templates/remix to RR v7

acusti committed:
Migrate apps/site to RR v7

acusti committed:
Update lock file with RR v7 dependency changes

acusti committed:
Remove defunct build.d.ts files, add GH env var

the placeholder GITHUB_TOKEN in .dev.vars ensures that wrangler includes it in its typegen (worker-configuration.d.ts)

acusti committed:
Remove deprecated @remix-run/eslint-config

acusti committed:
Upgrade @cloudflare/workers-types + tsup

acusti committed:
Adapt superflare from remix → react-router

this required duplicating node-adapter.js and maintaining it within superflare-remix, because it’s no longer included in the published package’s dist/ directory. here’s a PR that exposes the (from|to)NodeRequest utilities that we depend on in the superflareDevProxyVitePlugin: https://github.com/remix-run/react-router/pull/12774

if it gets merged, we can remove superflare-remix/node.adapter.ts and import those utils alongside cloudflareDevProxy

note that i also needed to update superflare-remix’s tsconfig.json module and moduleResolution settings to get it to build without error (presumably due to changes between the published remix vs react-router packages

acusti committed:
:memo: Update docs from remix → react-router

acusti committed:
Apply toJSON() to relations (stop password leak)

using serialize skips the potential toJSON() override used to e.g. not include User.password in the serialized form of a User when it is serialized as a relation of another model (e.g. Article in examples/remix-cms)

acusti committed:
Add single fetch serialization ownKeys proxy trap

single fetch uses turbo-stream to serialize data, meaning that model.toJSON() is no longer automatically invoked as was the case with the remix json() helper.

turbo-stream’s encode function calls flatten, which defines a partsForObj util that gets called for POJOs and uses Object.keys(): https://github.com/jacob-ebey/turbo-stream/blob/main/src/flatten.ts#L50

customizing ownKeys() to use toJSON() to return the appropriate keys ensures that the object gets serialized by turbo-stream in the form defined by the model ( including any customizations, e.g. the default User model, which removes the password field from the result)

however, calling target.toJSON() means that relation field keys are also returned, which causes the relation model name to exist on the model (though undefined). this is why we need to check in the get() proxy trap if the prop is not undefined on target, rather than just check for the existence of the prop in target.

acusti committed:
Add explicit toJSON() method to queries

allows for e.g.:

const articles = await Article.with("user").orderBy("createdAt", "desc").toJSON();

from a loader to explicitly return the serialized version of a model to provide it to a component. this makes it so that the version of the model that is provided during SSR is exactly equivalent to the version of the model that is made available to the component when rendered on the client. otherwise, the data made available to the component during SSR when using single fetch is the actual model instance.

acusti committed:
Update unused pages function example to latest API

fixes 4 type errors in examples/remix-cms

acusti committed:
📝 Fix typos/grammar in docs and comments

acusti committed:
Refactor Model+QueryBuilder types for single fetch

• add an IsSingle type argument to QueryBuilder to track if first() has been invoked
• add the toJSON method to QueryBuilder
• update return types of BaseModel query methods to return a QueryBuilder
• update QueryBuilder promise methods (then/catch) to resolve query results based on if IsSingle (i.e. if first() has been invoked) or not

acusti committed:
Remove relation promise union → improve Model type

this allows proper type checking of resolved instances of the model when using single fetch

acusti committed:
Remove relation promise union → improve Model type

this allows proper type checking of resolved instances of the model when using single fetch

jplhomer commented:
Sweet! Could we potentially use https://github.com/mjackson/remix-the-web/tree/main/packages/node-fetch-server#low-level-api for the `Request` generation, and https://github.com/mjackson/remix-the-web/tree/main/packages/headers for Headers conversion instead?

acusti committed:
Add react-router to typegen command

acusti committed:
Migrate docs site/remix-cms/template to RR types

acusti committed:
Return seed callback to enable awaiting result

ensures that the seed command can complete all of its async work and exit properly and makes it so that errors can be caught and logged (lines 45–47)

acusti committed:
Bump eslint(-register) dependencies

acusti committed:
Use @mjackson/node-fetch-server for viteDevServer

acusti committed:
Adopt latest entry.(client|server).tsx examples

• update entry.server.tsx based on: https://github.com/remix-run/react-router-templates/blob/main/cloudflare/app/entry.server.tsx
• remove entry.client.tsx files (they aren’t in the template examples any more)

acusti committed:
Update publish-snapshot.yml to get latest corepack

acusti commented:
@jplhomer good call! i switched over to `@mjackson/node-fetch-server` in c2512bb0a7ebf44cf9674708c6fd7fa286cd6352, which uncovered https://github.com/remix-run/remix/issues/10014. i took a look at the state-of-the-art for the [cloudflare templates](https://github.com/remix-run/react-router-templates/tree/main/cloudflare/app) in react-router-templates, and i saw that `entry.server.tsx` has a few significant changes (which resolved the newly uncovered AbortController issue) and that they have eliminated `entry.client.tsx`, so i made the same changes here in ab27d2e.

acusti committed:
Upgrade wrangler dependencies to v4

acusti committed:
Add wrangler.json $schema + required database_name

acusti committed:
Filter all internal cloudflare D1 tables (_cf_*)

new versions of wrangler include a new _cf_METADATA table that needs to be filtered out alongside _cf_KV, which i extrapolate to mean that any internal cloudflare tables introduced in the future will start with the _cf_ prefix
